pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -167,18 +167,80 @@ func extractMarker(comment *ast.Comment) Marker {
 		return Marker{}
 	}
 
+	markerContent := strings.TrimPrefix(comment.Text, "// +")
+	id, expressions := extractMarkerIdAndExpressions(markerContent)
+
 	return Marker{
-		Value:      strings.TrimPrefix(comment.Text, "// +"),
-		RawComment: comment.Text,
-		Pos:        comment.Pos(),
-		End:        comment.End(),
+		Identifier:  id,
+		Expressions: expressions,
+		RawComment:  comment.Text,
+		Pos:         comment.Pos(),
+		End:         comment.End(),
+	}
+}
+
+func extractMarkerIdAndExpressions(marker string) (string, map[string]string) {
+	// if there is only a single "=" split on the equal sign and trim any
+	// dangling ":" characters.
+	if strings.Count(marker, "=") == 1 {
+		splits := strings.Split(marker, "=")
+		// Trim any dangling ":" characters on the identifier to handle
+		// cases like +kubebuilder:object:root:=true
+		identifier := strings.TrimSuffix(splits[0], ":")
+
+		// If there is a single "=" sign that means the left side of the
+		// marker is the identifier and there is no real expression identifier.
+		// For now, set the expression identifier as an empty string.
+		// TODO: Is this sufficient? Can this be improved?
```

Comment:
Perhaps we can create a constant (that is actually an alias to the empty string) that makes this more obvious 🤔